test/src/test/java/hudson/model/queue/WideExecutionTest.java

```diff
@@ -87,6 +95,33 @@ public String getDisplayName() {
         }
     }
 
+    @Issue("JENKINS-30084")
+    @Test
+    /*
+     * this is to test that when the assigned executor is not available the flyweighttask is put into the buildable list,
+     * thus the node will be provisioned.
+     * when the flyweight task is not assigned to an offline executors the buildable list is empty.
+     *
+     */
+    public void flyWeightTaskQueue () throws IOException {
```

Comment:
DummyCloudImpl dummyCloud = new DummyCloudImpl(r, 0);
    dummyCloud.label = LabelExpression.get("aws-linux-dummy");
    r.getInstance().clouds.add(dummyCloud);
    project.setAssignedLabel(LabelExpression.get("aws-linux-dummy"));
    r.assertBuildStatusSuccess(project.scheduleBuild2(0));


The above waits until the build is succesfull which means that the slave is provisioned. Running the same test in the previous implementation of the code will fail with timeout since the node will not be provisioned and the project never executed